azure-quantum/azure/quantum/optimization/term.py

```diff
@@ -4,23 +4,45 @@
 ##
 
 from typing import List, Union, Optional
+import numpy as np
```

Comment:
Does this add numpy as a dependency?

Reply:
Yes, after following up with you offline, we're revisiting the approach for this change.

Reply:
That can be mitigated by guarding with an `if typing.TYPE_CHECKING` statement: https://docs.python.org/3/library/typing.html#typing.TYPE_CHECKING

Reply:
Per discussion offline, we're leaving the numpy dependency in, however, we check for numpy.typing conditionally to support this logic.

Reply:
Fair enough! NumPy is _extremely_ common in scientific Python audiences (may be nearly universal, even), but perhaps less so outside of scientific usage. If we believe that the primary users are likely to be scientific in nature, taking a NumPy dependency is nearly zero-weight as a result.